tools/building.py

```diff
@@ -684,7 +684,7 @@ def check_closure_compiler(cmd, args, env, allowed_to_fail):
     if isinstance(e, subprocess.CalledProcessError):
       sys.stderr.write(e.stdout)
     sys.stderr.write(str(e) + '\n')
-    exit_with_error('closure compiler ("%s --version") did not execute properly!' % str(cmd))
+    exit_with_error('closure compiler (%s) did not execute properly!' % str(cmd))
```

Comment:
You can use `shared.shlex_join` here rather than str.  Its job is to turn command line args into strings to echo'ing to the user.

Reply:
I think it is better to print in `['', '']` to capture issues with incorrect ordered parameters e.g. `['C:\Program', 'Files']`, which shell joining will hide.

Reply:
shlex_join would not know intent? It cannot understand that passing `['C:\Program', 'Files']` is wrong, but some calling code would have wanted to pass `['C:\Program Files']` instead? shlex_join would merge `['C:\Program', 'Files']` to `C:\Program Files`, and it can be difficult for the reader to know whether the printout intended to quote spaces with "s or to omit them.

Reply:
One of the key utilities of using `shared.shlex_join` when echoing is that a user/developer can copy and paste the output directly into their shell in order to reproduce the command that emscripten was trying to run.